optimade/models/baseinfo.py

```diff
@@ -100,6 +100,6 @@ def formats_and_endpoints_must_be_valid(cls, v, values):
 
 
 class BaseInfoResource(Resource):
-    id: str = Field("/", const="/", pattern="^/$")
-    type: str = Field("info", const="info", pattern="^info$")
+    id: str = Field("/", const="/", regex="^/$")
+    type: str = Field("info", const="info", regex="^info$")
```

Comment:
Are the `regex` change here with the validator in mind? If not, they are unnecessary.

Reply:
> I am not sure why you're getting the issues concerning the tests and `re' matching. But I've provided a suggested change for you to try out :)

:+1: 

> I wonder if we can do this in `Config` of a parent model instead, but perhaps not. Perhaps this is the best way. It's at least quite elegant in the end, I think. At least for what it is.

Yeah, I think the only other option is to define `OptimadeModel(BaseModel)` and `StrictModel(BaseModel)`, but I think that would involve a fair bit more hackery.

> Otherwise we should create a custom JSON encoder that can do all this and be used when creating the OpenAPI specification?

Yep, I think we have no other choice if we want this stuff in the schema.